backend/src/controllers/OrderController.ts

```diff
@@ -0,0 +1,111 @@
+import { Request, Response } from 'express';
+import Stripe from 'stripe';
+import Restaurant, { MenuItemType } from '../models/restaurant';
+
+const STRIPE = new Stripe(process.env.STRIPE_API_KEY as string);
+const FRONTEND_URL = process.env.FRONTEND_URL as string;
+
+type CheckoutSessionRequest = {
+  cartItems: {
+    menuItemId: string;
+    name: string;
+    quantity: number;
+  }[];
+  deliveryDetails: {
+    email: string;
+    name: string;
+    addressLine1: string;
+    city: string;
+  };
+  restaurantId: string;
+};
+
+const createCheckoutSession = async (req: Request, res: Response) => {
+  try {
+    const checkoutSessionRequest: CheckoutSessionRequest = req.body as CheckoutSessionRequest;
+    const restaurant = await Restaurant.findById(checkoutSessionRequest.restaurantId);
+    if (!restaurant) {
+      return res.status(404).send('Restaurant not found');
+    }
+
+    const lineItems = await createLineItems(checkoutSessionRequest, restaurant.menuItems);
+    const session = await createSession(
+      lineItems,
+      'TEST_ORDER_ID',
+      restaurant.deliveryPrice,
+      restaurant._id.toString(),
+    );
+
+    if (!session.url) {
+      return res.status(500).send('Error creating checkout session');
+    }
+
+    return res.status(200).json({ url: session.url });
+  } catch (error) {
+    console.log(error);
+    res.status(500).send('Error creating checkout session');
+  }
+};
+
+function createLineItems(
+  checkoutSessionRequest: CheckoutSessionRequest,
+  menuItems: MenuItemType[],
+) {
+  const lineItems = checkoutSessionRequest.cartItems.map((cartItem) => {
+    const menuItem = menuItems.find((item) => item._id.toString() === cartItem.menuItemId);
+    if (!menuItem) {
+      throw new Error('Menu item not found');
+    }
+
+    const lineItem: Stripe.Checkout.SessionCreateParams.LineItem = {
+      price_data: {
+        currency: 'gbp',
+        unit_amount: menuItem.price,
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/46#issuecomment-2423618327>#d63ac7</a><span> - 10/19/2024, 06:40 am</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The `unit_amount` is set directly to `menuItem.price`, which may not be in the correct format for Stripe (cents).
</div>
<div id="fix">
<b>Fix:</b> Convert the price to cents before assigning it to `unit_amount`. Multiply `menuItem.price` by 100 and round to the nearest integer.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -60,7 +60,7 @@
      const lineItem: Stripe.Checkout.SessionCreateParams.LineItem = {
        price_data: {
          currency: 'gbp',
 -        unit_amount: menuItem.price,
 +        unit_amount: Math.round(menuItem.price * 100),
          product_data: {
            name: menuItem.name,
          },
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged

Reply:
good observation, but i did it already